torch/_dynamo/config.py

```diff
@@ -401,6 +401,11 @@ def default_debug_dir_root():
 # WARNING: this is an experimental flag and is subject to change.
 _experimental_support_context_fn_in_torch_utils_checkpoint = False
 
+# Approximate maximum number of nodes to unroll loops into. A value of 0 will
+# unroll fully but could result in very large graphs that take an inordinate
+# amount of time to process.
+max_loop_unroll_nodes = int(os.environ.get("TORCHDYNAMO_MAX_LOOP_UNROLL_NODES", 5000))
```

Comment:
On the compile time improvements doc, I wrote down
```
[With Horace] Partial graphs
Instead of emitting one fully inlined graph, have dynamo emit particle graphs
For example, if you have a large for loop, instead of unrolling it, emit a partial graph for its contents, and re-use it. Balance this between compile time versus performance.
```

This sounds like a perfect example for when something like this would really benefitial.

@Chillee @anijain2305

Reply:
The same idea at https://github.com/pytorch/pytorch/issues/102839#issuecomment-1638873591